packages/react-ui-kit/src/Misc/useOutsideClick.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {RefObject, useEffect} from 'react';
+
```

Comment:
is there a reason we define this ourselves instead of using the go-to library -> https://github.com/streamich/react-use

It already has a solution for that (useClickAway) and will help us in different other cases as well (probably). Through tree shaking it is also not bigger than defining the hooks ourselves.

Reply:
Shouldn't our tree-shaking just use the singular function we use from this package and remove the rest of the code?